keps/sig-api-machinery/3488-cel-admission-control/README.md

```diff
@@ -1041,6 +1530,113 @@ representation. For admission control, we also need CEL to be integrated with
 the Kubernetes Go structs used to representative native API types, both for type
 checking and for runtime data access.
 
+### Writing to Status
+
+This enhancement proposes using status to of `ValidatingAdmissionPolicy` to
+communicate type-checking errors and any other misconfigurations such as CRD not
+found errors.
+
+As mentioned in
+https://github.com/kubernetes/enhancements/pull/3492#discussion_r964841045,
+status on API server configuration objects has been tricky to design in the
+past, because of the following:
+
+- multiple active kube-apiservers (sometimes at identical versions, sometimes
+  skewed by one version during upgrade)
+- multiple active non-kube-apiserver servers (aggregated servers)
+
+As a concrete example, The CRD `NonStructural` status field takes advantage of
+the metadata generation field
+(https://github.com/kubernetes/kubernetes/commit/2cfc3c69dc7c17b2711af0168f39ed7f515675c2).
+This avoids repeated updates for the same generation and potential fights of API
+server in HA environments even without leader elected controllers.
```

Comment:
If the controllers are not singletons, can there ever truly be one status?

Imagine a multi API server cluster is mid-upgrade and a resource gets created that leverages a field acknowledged in the newer version, but not the older version.

If 1/3 of the API servers are on the newer version, then that server will report "everything's looking good", the other two servers will report "error".

In this situation, enforcement will be random depending on which API server is hit, which is something a user may be interested in.

Reply:
The second half of this kubecon talk addresses status in a multi-controller/webhook environment:

https://sched.co/ekBY

Reply:
Not truely. 

And even when the controller is leader elected, it is going to write to status based on the Kubernetes version is communicates with so the same problem exists.

Even when all the apiservers are on the same version, one might be seeing a newer revision of a CRD than the others, resulting in a similar case.

Also, when a webhook is registered, the is no bound on how long it will take for all API servers to realize it exists and include it in the admission chain. And the apiserver currently make no attempts to show in the status of CRDs which apiservers they are registered (or which resourceVersion that apiserver is using). Same for webhooks.